app/knights/knight.py

```diff
@@ -0,0 +1,25 @@
+class Knight:
+
+    def __init__(self, config: dict) -> None:
+
+        self.name = config.get("name")
+        self.power = config.get("power")
+        self.hp = config.get("hp")
+        self.protection = 0
+
+        if config["armour"] and len(config["armour"]) > 0:
+            for item in config["armour"]:
+                self.protection += item["protection"]
+
+        if config["weapon"]["power"]:
+            self.power += config["weapon"]["power"]
+
+        if config["potion"] and config["potion"]["effect"]:

```

Comment:
The code assumes that 'potion' is always present in the config dictionary and that it contains an 'effect' key. Consider using `config.get('potion', {}).get('effect', {})` to avoid a `KeyError` if 'potion' or 'effect' is missing.